ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -203,12 +203,13 @@ let checkpoint ~__context ~vm ~new_name =
 		end;
 
 		(* snapshot the disks and the suspend VDI *)
-		let snap =
+		let snap, err =
 			if not (TaskHelper.is_cancelling ~__context) then begin
-				try Some (Xapi_vm_clone.clone Xapi_vm_clone.Disk_op_checkpoint ~__context ~vm ~new_name ~snapshot_info_record:!snapshot_info)
```

Comment:
As per CA-213568: Call `Xapi_vm_clone.clone Xapi_vm_clone.Disk_op_checkpoint ~__context ~vm ~new_name ~snapshot_info_record:!snapshot_info` fails with error `Server_error(SR_BACKEND_FAILURE_44, [ ; There is insufficient space;  ])`. This exception is raised before calling subsequent call `Xapi_xenops.resume ~__context ~self:vm ~start_paused:false ~force:false`. This is causing the VM to be in `suspended` state during a checkpoint.